src/typeguard.ts

```diff
@@ -0,0 +1,55 @@
+import { StaveNote } from 'stavenote';
+import { TabNote } from 'tabnote';
+
+/* eslint-disable */
+type Constructor<T> = Function & { prototype: T };
+
+/**
+ * Use this instead of `instanceof` as a more flexible type guard.
+ * @param cls a JavaScript class, such as `StaveNote`.
+ * @param obj check if this object is an instance of the provided `cls`.
+ * @param checkAncestors defaults to `true`, so we walk up the prototype chain to look for a matching `.getCategory()`.
+ *        If `false`, we do not check the superclass or other ancestors.
+ * @returns true if `obj` is an instance of `ClassName`, or has a `.getCategory()` that matches `ClassName.CATEGORY`.
+ */
+export function isCategory<Class>(cls: Constructor<Class>, obj: any, checkAncestors: boolean = true): obj is Class {
```

Comment:
Thanks Ron -- this change is great overall. QQ: why do wee need `cls` here? Can we not derive it from `Class`?

(Also, I prefer to call the template type `T` instead of `Class`.)

Reply:
Class is a typescript annotation, so it's removed in compilation.  It might be possible to do the opposite (get Class from cls) but I couldn't figure out a way to do it in current typescript.